### ISSUE TITLE
Trouble with ground targeted abilities

### ISSUE THREAD

**Issue by ATommasi** [RaidcoreGG/GW2-RadialMenus#2]
I created a new radial for the mastery skills. 
Things like the summon door and place skiff don't work well because you cannot move the ground targeting. They activate instantly instead of allowing you to place them. I couldn't find any workarounds

**Comment by DeltaGW2** [comment]
Depending on your in-game setting for Ground Targeting you want to set up the activation differently.
This is what I figured works best:

**Normal**
`InputBind (Game)` - This will trigger the Ground Targeting and you can click to drop it.

**Fast with Range Indicator** or **Instant**
`InputBind Press (Game)` - This will trigger the Ground Targeting and you can click to drop it.
`Delay` - Set to 5000ms to wait for you to position it.
`InputBind Release (Game)` - Even if you manually clicked to drop this action is needed to "unset" the pressed keys after 5 seconds.

In all fairness the second option works even if your Ground Targeting is set to Normal, but I find a direct Invoke smoother, more convenient personally.

**Comment by ATommasi** [comment]
Thanks! I was able to make a bit of progress but it's still not working as expected for me. 

I have Fast w/Range Indicator set.
I have all my mounts as numpad 1-9 and my mastery skills as alt-shift-numpad 1-6

Here's why I've done.
I changed Radial settings to use InputBInd Press (Game) and it wasn't working still

If I used the mastery radial to trigger alt-shift-numpad(1)  I would place a door and mount up at the same time, like it was ignoring the modifier keys. I then changed my conjure doorway to be completely unique and set it to Ctrl-Alt-[period] and it worked. But only once. After that, activating the radial item does nothing until I change the keybind again. Then it'll work once and not again after. I've tried a few keybinds and am getting the same behavior consistently, even using keybinds without modifiers.

Hopefully I explained that clearly enough. Thanks so much for this awesome addon though, it's been working great otherwise!

Also, I'm not sure if it matters, but I am running Linux as my OS

**Comment by ATommasi** [comment]
*** UPDATE *** 
I found that after it works the first time, I don't have to change the keybinds to get it to work again. I just need to wait 30-40 seconds. 

Or I can manually press the keys on my keyboard. The first time pressing does nothing, but the second time it works. Then after that the radial will work again. It's almost like InputBind Press isn't releasing the keybind and eventually times out.
Manually pressing the keys on the keyboard seems to force the full key press event to finish allowing the radial to use it again.

**Comment by DeltaGW2** [comment]
> InputBind Press isn't releasing the keybind
That's correct, InputBind Press merely sends the Press, not the Release. The action called just `InputBind (Game)` handles both pressing and releasing.
Press and Release exist as separate actions for more advanced use-cases like dodge jumping.

**Comment by ATommasi** [comment]
That makes sense. 

I switched back to using Just InputBind and now the game crashes everytime I use either skiff or door. The error inidicates my system ran out of memory.. so possible memory leak somewhere? This wasn't happening before so I don't know what changed.

 There's definitely issues with ground targeting abilities. I just can't tell if it's me or the addon. Could very well be me.

And the issue with the modifiers still allowing the base key press to pass through seems to still be happening. Num(1) and Alt-shit-num(1) both trigger the num(1) assignment

Thanks for taking the time with all of this. And I apologize if some of these issues are just my own ignorance. If there's anything I can do to help, let me know. I am a software developer by trade, but I deal mainly with web apps.

**Comment by DeltaGW2** [comment]
If you have a crashlog (from ArcDPS) it'd be good to see. (Nexus own crashlogger is not working as intended and is being rewritten.)

Ground Targeting as far as I've tested as described in the previous comment works fine, it requires tweaking to fit the settings but works fine in that configuration.

As for the Num 1 pass-through, I'm not entirely sure how that is happening or if I understand what you mean.
Could you elaborate, please?

Don't ever worry about asking, worst case (for you) it's a quick answer to get a solution, worst case (for me) is finding a bug I can fix!

**Comment by ATommasi** [comment]
I'll see if I can find a crashlog for you. Thanks for your patience and for making this awesome addon. I'm sure the issues I'm having are unique to my system and I appreciate the effort you put in.

**Comment by ATommasi** [comment]
Here is the last crash log I have
[arcdps_lastcrash.log](https://github.com/user-attachments/files/17002935/arcdps_lastcrash.log)

**Comment by ATommasi** [comment]
I was finally able to get everything to work. I completely removed the radial menu I created for masteries and recreated it. Something must have gotten messed up in there that was causing my crashes, becuase it doesn't happen anymore.

As for the numpad passthhrough issue, it's basically this:
I have all my mounts assigned to numpad(1-9)
My masteries were assigned to  alt-shift-numpad(1-6)

Activating masteries caused both the mount and the mastery to trigger. Changing my mastery keybinds to use  ctrl-alt-numpad(x) instead of shift-alt resolved this somehow. So it seems using the shift key as a modifier doesn't work? At least in my setup for some reason.

The second method you suggested above is now working as intended.

**Comment by ATommasi** [comment]
And in case anybody else wants it, I've attached the mastery icons and menu here
[Mastery_Radial_Menu.zip](https://github.com/user-attachments/files/17003021/Mastery_Radial_Menu.zip)

**Comment by DeltaGW2** [comment]
Hmm, that does sound a bit weird, it's usually Alt causing problems since that's a SYS key. It might be a priority issue on Linux? I will investigate, thanks for sharing the log and the radial.

There's one rare crash where calling a radial from a radial will cause the game to crash, and I'm fixing that by implementing special behaviour for that case.
